src/main/app/controller/BasePaymentPostController.ts

```diff
@@ -30,19 +28,26 @@ export default abstract class BasePaymentPostController {
     if (req.session.userCase.state !== this.awaitingPaymentState()) {
       req.session.userCase = await req.locals.api.triggerEvent(
         req.session.userCase.id,
-        {},
+        { citizenPaymentCallbackUrl: getPaymentCallbackUrl(req, res) },
         this.awaitingPaymentEvent()
       );
     }
 
-    const payments = new PaymentModel(req.session.userCase[this.paymentsCaseField()] || []);
+    if (!this.getServiceReferenceForFee(req)) {
+      req.session.userCase = await req.locals.api.triggerEvent(
+        req.session.userCase.id,
+        { citizenPaymentCallbackUrl: getPaymentCallbackUrl(req, res) },
+        CITIZEN_CREATE_SERVICE_REQUEST
+      );
+    }
```

Comment:
We can remove this once any cases that were in AwaitingPayment or AwaitingFinalOrderPayment at the time of release have progressed, as the service request ref will be set with the order summary in CITIZEN_SUBMIT/RESPONDENT_APPLY_FINAL_ORDER for new cases